lib/fs.js

```diff
@@ -2051,7 +2053,29 @@ function writeAll(fd, isUserFd, buffer, offset, length, signal, callback) {
     }
     return;
   }
-  // write(fd, buffer, offset, length, position, callback)
+  if (isCustomIterable(buffer)) {
+    (async () => {
+      for await (const buf of buffer) {
+        fs.write(
+          fd, buf, undefined,
+          isArrayBufferView(buf) ? buf.byteLength : encoding,
+          null, (writeErr, _) => {
+            if (writeErr) {
+              if (isUserFd) {
+                callback(writeErr);
+              } else {
+                fs.close(fd, (err) => {
+                  callback(aggregateTwoErrors(err, writeErr));
+                });
+              }
+            }
+          }
+        );
+      }
+      fs.close(fd, callback);
```

Comment:
I can't find a good solution to the following problems...
I would like to ask for your confirmation.

- Occasionally, The order of characters to be written is not correct.
- Occasionally, Not all characters are written (this can be solved by delaying the close after the `for await of` statement as shown below)
```suggestion
     setTimeout(() => {
       fs.close(fd, callback);
     }, 100)
```

Reply:
Currently you are both writing and reading from the buffer "at the same time", as `fs.write` is non-blocking and will write in parallel for chunks as they arrive. I think that either promisifying the fs.write and awaiting it (which feels a bit wrong for me, in the callback version), or maybe iterating yourself manually (using .`next()` and then with callback) is the more correct path here. 

Also note that you might end up calling `close` twice here (if an error occured and at the end of the buffer), and also closing it prematurely if the iterator has ended but the previous `write` is still happening.